core/server/master/src/main/java/alluxio/master/file/MetadataSyncLockManager.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file;
+
+import alluxio.AlluxioURI;
+import alluxio.collections.LockPool;
+import alluxio.concurrent.LockMode;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.exception.InvalidPathException;
+import alluxio.resource.LockResource;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.base.Preconditions;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Class for managing metadata　sync locking.
+ */
+public class MetadataSyncLockManager {
+  /**
+   * Pool for supplying metadata sync locks.
+   *
+   * We use weak values so that when nothing holds a reference to
+   * a lock, the garbage collector can remove the lock's entry from the pool.
+   */
+  private final LockPool<String> mLockPool =
+      new LockPool<>((key) -> new ReentrantReadWriteLock(),
+          Configuration.getInt(PropertyKey.MASTER_LOCK_POOL_INITSIZE),
+          Configuration.getInt(PropertyKey.MASTER_LOCK_POOL_LOW_WATERMARK),
```

Comment:
these watermarks are good choices for master lock pool, are they too large for this?

Reply:
yeah probably let's reduce the size